src/IonParserBinaryRaw.ts

```diff
@@ -331,24 +337,31 @@ export class ParserBinaryRaw {
                     throw new Error('Timestamps with an hour must include a minute.');
                 } else {
                     minute = this.readVarUnsignedInt();
-                    precision = Precision.HOUR_AND_MINUTE;
                 }
+                precision = Precision.HOUR_AND_MINUTE;
             }
             if (this._in.position() < end) {
-                second = this.readVarUnsignedInt();
+                secondInt = this.readVarUnsignedInt();
+                second = new Decimal(new LongInt(secondInt), 0);
                 precision = Precision.SECONDS;
             }
-            if(this._in.position() < end) {
+            if (this._in.position() < end) {
                 let exp = this.readVarSignedInt();
                 let coef = LongInt._ZERO;
-                if(this._in.position() < end) {
+                if (this._in.position() < end) {
                     coef = ParserBinaryRaw.readSignedIntFrom(this._in, end - this._in.position());
                 }
-                fraction = new Decimal(coef, exp);
-                precision = Precision.FRACTION;
+                let decimalStr = secondInt.toString() + coef.toString() + 'd' + exp;
```

Comment:
This block of code is a huge performance hit, these tostring apis are base going from base 128 to base 1e7 to base 10 to generate the string and back again. I would block the pr based on these lines.

Reply:
We own `Decimal`, but this case doesn't fit its (coefficient, exponent) constructor, as the desired coefficient is a combination of `secondInt` and `coef`.

Agreed that it is round-about;  I've also realized it is incorrect:  you can't just concat secondInt and the decimal coefficient (otherwise a value that should be `5.00123` is read as `5.05123`).  I'm a bit surprised none of the tests have caught this...I'll address this shortly.

If Decimal had an `add()` method, it would be more natural to call `new Decimal(coef, exp)`, then add `secondInt`, but we're intentionally avoiding support for Decimal math operations at this time.  Open to better ideas....

Reply:
Ok, I've added a regression test to catch this bug.

As far as performance, our focus is on correctness and providing the right and minimal API surface area.  Performance is likely to become higher priority later.

Reply:
i don't believe the refactor is related to correctness, not sure i follow the logic.

Reply:
I benched it and its only 2-4x slower, not a blocker at all.